src/lua_details.cc

```diff
@@ -54,6 +56,14 @@ static const char step_timeout_s_key[] =
 
 namespace task {
 
+// A holder for a sequence timeout pointer to trigger an abort when the clock elapsed.
+TimeoutTrigger* sequence_timeout_{nullptr};
```

Comment:
This is a global variable? I guess that won't work at all when running multiple sequences in parallel.

Reply:
Good point! Because if we run multiple sequence consecutively and every sequence sets its own timeout the last one wins. Therefore your approach about adding a `Sequence` pointer to`Step::execute(...)` becomes more or less the only possibility to solve the problem (per default set to `nullptr`). `Step::execute(...)` now forwards the sequence pointer to the Lua hook which is now 'thread-safe'. However, we have a circular dependency of `Sequence` in `Step` which I don't like  ... :-/ Anyhow I add a use case to sequence test that tests the timeout on two parallel sequences during execution.

Reply:
Yup! This resolves the circular dependency :smile:.

Any additional tweaks/enhancements...